.github/workflows/lint.yml

```diff
@@ -0,0 +1,19 @@
+name: lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        
```

Comment:
Oh and we're missing the golangci-lint version. We want to define it so that the action is deterministic, otherwise running it tomorrow might fail if the version is updated.